packages/tools/src/tools/displayTools/Labelmap/labelmapDisplay.ts

```diff
@@ -134,6 +135,61 @@ function removeSegmentationRepresentation(
   }
 }
 
+/**
+ * Checks if a segmentation data belongs to the same study as the series
+ * displayed in a given viewport
+ * @param viewport
+ * @param referencedVolumeId volume id of the segmentation reference series
+ * @returns
+ */
+function canRenderSegmentationInViewport(
+  viewport,
+  referencedVolumeId,
+  restrictVisualization
+) {
+  const defaultActor = viewport.getDefaultActor();
+  if (defaultActor) {
+    const { uid: defaultActorUID } = defaultActor;
+    const volume = cache.getVolume(defaultActorUID);
+    const viewportImageIds = volume.imageIds;
+
+    if (viewportImageIds) {
+      const referencedVolume = cache.getVolume(referencedVolumeId);
+      const segmentationImageIds = referencedVolume.imageIds;
+      if (segmentationImageIds) {
+        const segmentationSeriesInfo = metaData.get(
+          'generalSeriesModule',
+          segmentationImageIds[0]
+        );
+        const segmentationImageInfo = metaData.get(
+          'imagePlaneModule',
+          segmentationImageIds[0]
+        );
+        const viewportSeriesInfo = metaData.get(
+          'generalSeriesModule',
+          viewportImageIds[0]
+        );
+        const viewportImageInfo = metaData.get(
+          'imagePlaneModule',
+          viewportImageIds[0]
+        );
+        if (
+          segmentationSeriesInfo.studyInstanceUID ===
+            viewportSeriesInfo.studyInstanceUID &&
+          viewportImageInfo.frameOfReferenceUID ===
```

Comment:
The same FOR is an absolute requirement to visualize the segmentation, so that one shouldn't be occurring now.  If it is, then it is an outright defect and needs to be addressed independently of a flag.

Reply:
This isn't dependent on D&D, but consider a series with images from different FOR's (forget about the fact right now that we can't render on non-volumes), if you navigate through that series, the different segmentations should apply automatically.
Similarly, if you just load 2 segmentations, and then display a single series, if the FOR is not the segmentation one, then the segmentation must not be displayed because the coordinates don't match.  That is exactly the situation:
There is a mirror 3 steps past the doorway at  123 My Street
and then expecting to find a mirror at 3 steps past he doorway at 456 Your Street.
(mirror at is the segmentation, and the FOR is 123 My Street or 456 Your Street).
I may well load information on both places, but if the main images aren't from either segmentation, then neither applies.